.github/workflows/convert-cci-list.yml

```diff
@@ -0,0 +1,60 @@
+name: Convert CCI List XML to JSON
+
+on:
+  push:
+    branches: ['master']
+
+  # Run this workflow on the 1st day at 00:00 every month
+  schedule:
+    - cron: '0 0 1 * *'
+
+env:
+  # This URL is super brittle with how links constantly get changed.
+  CCI_LIST_ZIP_URL: https://dl.dod.cyber.mil/wp-content/uploads/stigs/zip/U_CCI_List.zip
+
+jobs:
+  convert-cci-list:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the code
```

Comment:
https://github.com/mitre/redhat-enterprise-linux-9-stig-baseline/blob/main/.github/workflows/verify-ec2.yml#L38-L43

<html><head></head><body><div><div jsname="OsZLCd" data-mrc="GAE" class="yf" data-async-fc="EqEBCmJBSkc5SmtNczQ3YmdVY084WEFLVU54QnpoQTVMN1VaTG1ReGNDNmlTLUxYSnFnb0ZNLUN0Vmp1Q09WYmhscjF0X2lFSWJQazhQVklQc1EtRmFGSWdOZkwzWlV1aTl3OTYxZxIXakhZOVo0V09DSmFpNU5vUGdfT2h5QXMaIkFGWHJFY29QZFQzbmpkNU5ZZjFDb1UyZ1N3UE1iaWpHdHc" data-async-fcv="3" data-async-ons="7159" id="fc_jHY9Z4WOCJai5NoPg_OhyAs_1" eid="23Y9Z7yZOpue5NoPxPjn4Qk"><div jsname="OsZLCd" data-mdv="0" data-mrc="GAE" class="yp" data-async-fc="EqEBCmJBSkc5SmtOVFBTbk42aEszejc2dGdOTHV2OXROeGZBTW1Mb3dTZzN1NXhuQXpiZWJMdXBGRnRXMkFHNDdrZldaeXZQODRpQ0pJcllseTE4VkdjVXd4ZE92SEloOU1SMGU3QRIXMjNZOVo3eVpPcHVlNU5vUHhQam40UWsaIkFGWHJFY3FqRkswbEJCWXY3OTN6QXY0czZLODFCOG55WVE" data-async-fcv="3" data-async-ons="7159" id="fc_23Y9Z7yZOpue5NoPxPjn4Qk_1"><div><div jsname="GVaozf" style="font-size:11pt"><div data-di="0"><div><span>In a GitHub Action workflow, the command git fetch --prune --unshallow is used to retrieve the latest code from a remote repository, removing any outdated remote branches and fully fetching the repository history (if it was previously a shallow clone) before pushing changes with git push, effectively ensuring your action has the most up-to-date code to work with. [<a href="https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead." data-ic="">1</a>, <a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">3</a>, <a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">4</a>]  </span></div><br style="line-height:11pt"><div style="font-size:13pt;font-weight:bold;margin-bottom:11pt"><span>Breakdown of the command: [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">3</a>, <a href="https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch." data-ic="">5</a>]  </span></div><ul><li><span>git fetch: </span><span>This command retrieves the latest commits from the remote repository without merging them into your local working directory. [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">3</a>, <a href="https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch." data-ic="">5</a>]  </span></li><li><span>--prune: </span><span>This option removes any remote-tracking branches that no longer exist on the remote repository, essentially cleaning up your local repository. [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://github.com/actions/checkout/issues/206" data-ic="">6</a>, <a href="https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file." data-ic="">7</a>]  </span></li><li><span>--unshallow: </span><span>If your local repository was previously a shallow clone (only containing a limited commit history), this flag will fetch the full history of the repository. [<a href="https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead." data-ic="">1</a>, <a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">4</a>]  </span></li></ul><div><span>How to use in a GitHub Action: </span></div><br style="line-height:11pt"><div data-cbl="" data-cbc="">
jobs:   build:     runs-on: ubuntu-latest     steps:       - uses: actions/checkout@v3       - run: git fetch --prune --unshallow       - run: # Your build commands here       - run: git push origin HEAD:your-branch
--


<br style="line-height:11pt"><div data-cbi=""><div data-cob=""></div></div></div><div style="font-size:13pt;font-weight:bold;margin-bottom:11pt"><span>Key points: [<a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">4</a>]  </span></div><ul><li><span><span style="font-weight:bold">When to use:</span> </span><span>This is particularly important when working with large repositories where a shallow clone might be used initially to save time and bandwidth. [<a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">4</a>]  </span></li><li><span><span style="font-weight:bold">Benefits:</span> [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">3</a>, <a href="https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch." data-ic="">5</a>]  </span><ul><li><span>Ensures you have the latest code from the remote repository. [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">3</a>, <a href="https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch." data-ic="">5</a>]  </span></li><li><span>Prevents clutter by removing outdated remote-tracking branches. [<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://github.com/actions/checkout/issues/206" data-ic="">6</a>, <a href="https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file." data-ic="">7</a>]  </span></li><li><span>Allows for full access to the commit history if needed. [<a href="https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead." data-ic="">1</a>, <a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">2</a>, <a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">4</a>]  </span></li></ul></li></ul><br style="line-height:11pt"><div><div style="margin-bottom:11pt"><i>Generative AI is experimental.</i></div></div><div data-sf="">[1]&nbsp;<a href="https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead." data-ic="">https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd</a><br></div><div data-sf="">[2]&nbsp;<a href="https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository." data-ic="">https://www.atlassian.com/git/tutorials/git-prune</a><br></div><div data-sf="">[3]&nbsp;<a href="https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches." data-ic="">https://www.atlassian.com/git/tutorials/syncing/git-push</a><br></div><div data-sf="">[4]&nbsp;<a href="https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos." data-ic="">https://graphite.dev/guides/git-shallow-clone</a><br></div><div data-sf="">[5]&nbsp;<a href="https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch." data-ic="">https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository</a><br></div><div data-sf="">[6]&nbsp;<a href="https://github.com/actions/checkout/issues/206" data-ic="">https://github.com/actions/checkout/issues/206</a><br></div><div data-sf="">[7]&nbsp;<a href="https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file." data-ic="">https://git-scm.com/docs/git-fetch</a><br></div><br style="line-height:11pt"></div></div></div></div></div></div></body></html>In a GitHub Action workflow, the command git fetch --prune --unshallow is used to retrieve the latest code from a remote repository, removing any outdated remote branches and fully fetching the repository history (if it was previously a shallow clone) before pushing changes with git push, effectively ensuring your action has the most up-to-date code to work with. [[1](https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead.), [2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [3](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.), [4](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)]  
Breakdown of the command: [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [3](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.), [5](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch.)]  

• git fetch: This command retrieves the latest commits from the remote repository without merging them into your local working directory. [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [3](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.), [5](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch.)]  
• --prune: This option removes any remote-tracking branches that no longer exist on the remote repository, essentially cleaning up your local repository. [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [6](https://github.com/actions/checkout/issues/206), [7](https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file.)]  
• --unshallow: If your local repository was previously a shallow clone (only containing a limited commit history), this flag will fetch the full history of the repository. [[1](https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead.), [2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [4](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)]  

How to use in a GitHub Action: 
jobs:

  build:

    runs-on: ubuntu-latest

    steps:

      - uses: actions/checkout@v3

      - run: git fetch --prune --unshallow

      - run: # Your build commands here

      - run: git push origin HEAD:your-branch


Key points: [[4](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)]  

• When to use: This is particularly important when working with large repositories where a shallow clone might be used initially to save time and bandwidth. [[4](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)]  
• Benefits: [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [3](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.), [5](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch.)]  
	• Ensures you have the latest code from the remote repository. [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [3](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.), [5](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch.)]  
	• Prevents clutter by removing outdated remote-tracking branches. [[2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [6](https://github.com/actions/checkout/issues/206), [7](https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file.)]  
	• Allows for full access to the commit history if needed. [[1](https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead.), [2](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.), [4](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)]  




Generative AI is experimental.

[1] [https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd](https://dev.to/camptocamp-ops/unshallowing-a-git-repository-24nd#:~:text=Git%20provides%20a%20fetch%20%2D%2Dunshallow%20command%20which,branch%2C%20so%20we%20need%20fetch%20%2D%2Dall%20instead.)
[2] [https://www.atlassian.com/git/tutorials/git-prune](https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is%20the%20best%20utility,longer%20in%20use%20on%20the%20remote%20repository.)
[3] [https://www.atlassian.com/git/tutorials/syncing/git-push](https://www.atlassian.com/git/tutorials/syncing/git-push#:~:text=The%20git%20push%20command%20is%20used%20to,branches%2C%20pushing%20exports%20commits%20to%20remote%20branches.)
[4] [https://graphite.dev/guides/git-shallow-clone](https://graphite.dev/guides/git-shallow-clone#:~:text=Unlike%20a%20regular%20clone%2C%20which%20downloads%20the,environments%20or%20when%20working%20with%20large%20monorepos.)
[5] [https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository](https://docs.github.com/en/get-started/using-git/getting-changes-from-a-remote-repository#:~:text=Use%20git%20fetch%20to%20retrieve%20new%20work,add%20a%20new%20remote%20and%20then%20fetch.)
[6] https://github.com/actions/checkout/issues/206
[7] [https://git-scm.com/docs/git-fetch](https://git-scm.com/docs/git-fetch#:~:text=By%20using%20a%20refspec%20that%20fetches%20tags,a%20remotes.%20entry%20in%20the%20configuration%20file.)

Reply:
I'm not sure if getting the full history of the repo is useful to the added GitHub Action (unless somehow the cciListXml2json.ts script is getting updated constantly)